tests/acceptance/test_update.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/python
+# Copyright 2016 Mender Software AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from fabric.api import *
+
+import pytest
+import unittest
+
+import os.path
+import subprocess
+
+# Make sure common is imported after fabric, because we override some functions.
+from common import *
+
+@pytest.mark.usefixtures("qemu_running", "no_image_file")
+class TestUpdates:
+    def test_broken_image_update(self):
+        if not env.host_string:
+            # This means we are not inside execute(). Recurse into it!
+            execute(self.test_broken_image_update)
+            return
+
+        # Make a dummy/broken update
+        run("dd if=/dev/zero of=image.dat bs=1M count=0 seek=8")
+        run("mender -rootfs image.dat")
+
+        try:
+            reboot()
+            # This should never be reached, because the reconnect should fail.
+            assert(False)
+        except:
+            pass
+
+        kill_qemu()
+        start_qemu()
+
+        output = run_after_connect("mount")
+
+        # The update should have reverted to /dev/mmcblk0p2, since the image was
+        # bogus.
+        assert(output.find("/dev/mmcblk0p2") >= 0)
+        assert(output.find("/dev/mmcblk0p3") < 0)
```

Comment:
Maybe we can make this configurable? If we will play with image creating Yocto recipes this might be different in future.